File: LinkedList/ReverseLinkedList.py
"""
Reverse the LinkedList
"""

class Node:
    def __init__(self,data,next):
        self.data = data
        self.next = next

class LinkedList:
    def __init__ (self):
        self.currnode = None

    def add(self,data):
        newNode = Node(data,self.currnode)
        self.currnode = newNode
        return self.currnode
    
    def _print(self):
        node = self.currnode
        while node:
            print node.data
            node = node.next
       
    def ReverseList(self):
        prev_node = None
        currentHead = self.currnode       
    
        while currentHead:
            copy_prev_node = prev_node
            copy_next_node = currentHead.next

            prev_node = currentHead
            currentHead.next = copy_prev_node
            currentHead = copy_next_node
    
        self.currnode = prev_node
        return prev_node
       
if __name__ == "__main__":
    li = LinkedList()
    li.add(2)
    li.add(4)
    li.add(5)
    li.ReverseList()
    li._print()

File: LinkedList/ReverseLinkedList.js
//Reverse the LinkedList

function Node(data, next) {
    this.data = data;
    this.next = next;
}

function LinkedList() {
    this.currnode = null;
}

LinkedList.prototype = {
    add: function(data) {
        var newNode = new Node(data, this.currnode);
        this.currnode = newNode;
    },
    printall: function() {
        var head = this.currnode;
        while (head != null) {
            console.log(head.data);
            head = head.next;
        }
    },
    reverseList: function() {
        var prev_head = null;
        var currentHead = this.currnode;
        while (currentHead != null) {
            var copy_prev_head = prev_head;
            var copy_currentHead_next = currentHead.next;

            prev_head = currentHead;
            currentHead.next = copy_prev_head;
            currentHead = copy_currentHead_next;
        }
        this.currnode = prev_head;
    }
}

var li = new LinkedList();
li.add(2);
li.add(4);
li.add(6);
li.reverseList();
li.printall();